File: SharpBlitz.Compiler.Abstractions/CompilationInput.cs
using SharpBlitz.Common.Assembly;
using System;
using System.Collections.Generic;

namespace SharpBlitz.Compiler.Abstractions
{
    public class CompilationInput
    {
        public string AssemblyName { get; set; } = Guid.NewGuid().ToString() + ".dll";
        public IEnumerable<string> Sources { get; set; }
        public bool DebugMode { get; set; } = true;
        public ProgramType ProgramType { get; set; } = ProgramType.ConsoleApp;
        public IEnumerable<AssemblyDefinition> AdditionalReferences { get; set; }
    }

    public enum ProgramType
    {
        ConsoleApp = 0,
        ClassLibrary = 1
    }
}

File: SharpBlitz-UI/src/app/models/compilation-input.ts
import { AssemblyDefinition } from './assembly-definition';

export interface CompilationInput {
  assemblyName: string;
  sources: string[];
  debugMode: boolean;
  programType: ProgramType;
  additionalReferences?: AssemblyDefinition[];
}

export enum ProgramType {
  ConsoleApp = 0,
  ClassLibrary = 1
}
